### ISSUE TITLE
[Feature] Add namePattern / filenamePattern fields to match against filename only (not full path)

### ISSUE THREAD

**Issue by fabianvf** [konveyor/analyzer-lsp#1100]
## Background

The `builtin.file` condition's `pattern` field and the `builtin.filecontent` condition's `filePattern` field currently match against the **full file path**, not just the filename (basename). This causes two classes of bugs:

1. **False positives** — A pattern like `hibernate.*\.jar` intended to match a JAR file will also match if a *directory* in the path happens to contain "hibernate" (e.g. `/path/to/hibernate-rules/unrelated.txt`).
2. **False negatives** — A `filePattern` exclude like `test` will filter out files simply because the *repository path* contains "test" (e.g. `/home/user/test-project/src/Main.java`).

### Root cause

In [`provider/lib.go:235`](https://github.com/konveyor/analyzer-lsp/blob/main/provider/lib.go#L235):

```go
regex.MatchString(file) || regex.MatchString(filepath.Base(file))
```

The pattern is tested against the full `file` path first. Even though it also tests `filepath.Base(file)`, the full-path match fires first and produces incorrect results when directory names coincidentally match.

## Prior Issues

- **[#1092](https://github.com/konveyor/analyzer-lsp/issues/1092)** — `builtin.file` `pattern` matches directory names, producing false positives
- **[#1083](https://github.com/konveyor/analyzer-lsp/issues/1083)** — `filePattern` fails when the repo path itself matches an exclude pattern, producing false negatives

## Evidence of Workarounds

The [Azure/appcat-konveyor-rulesets](https://github.com/Azure/appcat-konveyor-rulesets) team discovered this behavior and committed extensive workarounds across multiple PRs:

| PR | Description |
|---|---|
| [Azure#41](https://github.com/Azure/appcat-konveyor-rulesets/pull/41) | "Update rules description and regex" — documents the finding and applies workarounds across azure, openjdk, cloud-readiness, and os scopes |
| [Azure#42](https://github.com/Azure/appcat-konveyor-rulesets/pull/42) | "Update rules regex" — applies workarounds to technology-usage scope |
| [Azure#63](https://github.com/Azure/appcat-konveyor-rulesets/pull/63) | "Fix regex and add test for local storage rules" |
| [Azure#178](https://github.com/Azure/appcat-konveyor-rulesets/pull/178) | "fix Dockerfile file pattern match" |

From PR #41/#42, the contributor explicitly documented:
> - `filePattern` is the regex of **file path, which should be specified the path separator**
> - `pattern` (builtin.file) is the regex of **file name, which should be specified start point ^**

### Workaround pattern examples

**`filePattern` with path-separator anchoring** ([`03-azure-database-config.yaml` L58](https://github.com/Azure/appcat-konveyor-rulesets/blob/6778ffc0974164f3895971829db28f4b1ec523de/default/generated/azure/03-azure-database-config.yaml#L58)):
```yaml
# Workaround: (/|\\) forces a path separator before the filename
filePattern: ".*(/|\\\\)application(-[a-zA-Z0-9]+)?\\.(properties|yaml|yml)$"
```

**`pattern` matching full path** ([`127-embedded-framework-libraries.yaml` L12](https://github.com/Azure/appcat-konveyor-rulesets/blob/6778ffc0974164f3895971829db28f4b1ec523de/default/generated/technology-usage/127-embedded-framework-libraries.yaml#L12)):
```yaml
# Matches full path — risks false positives from directory names
pattern: ".*hibernate.*\\.jar$"
```

## Why We Should Not Change Existing Behavior

Users (notably the Azure/appcat-konveyor-rulesets team) have already adapted their rule patterns to account for the full-path matching semantics. Changing `pattern` or `filePattern` to match only the basename would **break these existing workarounds**. The path-separator patterns like `(/|\\\\)` would become meaningless, and carefully anchored patterns would stop matching.

## Proposed Solution

Add **new** fields that match only against `filepath.Base(file)` (the filename/basename):

| Condition | Existing field (full path) | New field (basename only) |
|---|---|---|
| `builtin.file` | `pattern` | **`namePattern`** |
| `builtin.filecontent` | `filePattern` | **`filenamePattern`** |

### Semantics

- `namePattern` / `filenamePattern` match the regex against `filepath.Base(file)` **only**
- The existing `pattern` / `filePattern` fields retain their current full-path behavior
- If both old and new fields are specified, both must match (AND semantics)

### Before / After Examples

**Matching application properties files:**
```yaml
# Before (workaround with path separator anchoring):
filePattern: ".*(/|\\\\)application(-[a-zA-Z0-9]+)?\\.(properties|yaml|yml)$"

# After (simple basename match):
filenamePattern: "application(-[a-zA-Z0-9]+)?\\.(properties|yaml|yml)$"
```

**Matching Dockerfiles:**
```yaml
# Before (workaround):
pattern: "(.*/)?Dockerfile.*"

# After (simple basename match):
namePattern: "Dockerfile.*"
```

**Matching JAR files:**
```yaml
# Before (full path match, risk of false positives):
pattern: ".*hibernate.*\\.jar$"

# After (only matches the actual filename):
namePattern: "hibernate.*\\.jar$"
```

## Implementation Notes

The change would be localized to `provider/lib.go` around line 235. When the new field is present, match only against `filepath.Base(file)` instead of the full path.

## Request for Discussion

Before implementing, we'd like feedback from stakeholders — especially the Azure/appcat-konveyor-rulesets team who have been most affected by this behavior:

- Are `namePattern` / `filenamePattern` the right field names?
- Are AND semantics the right choice when both old and new fields are specified?
- Are there use cases where basename-only matching would still be insufficient?

@haoozhang curious to hear your thoughts as it seems you implemented most of those workarounds

## Related

- #1092
- #1083
- [Azure/appcat-konveyor-rulesets PR #41](https://github.com/Azure/appcat-konveyor-rulesets/pull/41)
- [Azure/appcat-konveyor-rulesets PR #42](https://github.com/Azure/appcat-konveyor-rulesets/pull/42)
- [Azure/appcat-konveyor-rulesets PR #63](https://github.com/Azure/appcat-konveyor-rulesets/pull/63)
- [Azure/appcat-konveyor-rulesets PR #178](https://github.com/Azure/appcat-konveyor-rulesets/pull/178)
